File: sdk/csharp/libraries/microsoft.bot.builder.solutions/Authentication/ProviderTokenResponse.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System;
using Microsoft.Bot.Schema;

namespace Microsoft.Bot.Builder.Solutions.Authentication
{
    [Obsolete("This type is being deprecated. It's moved to the assembly Microsoft.Bot.Solutions. Please refer to https://aka.ms/botframework-solutions/releases/0_8", false)]
    public class ProviderTokenResponse
    {
        public OAuthProvider AuthenticationProvider { get; set; }

        public TokenResponse TokenResponse { get; set; }
    }
}

File: sdk/typescript/libraries/botbuilder-solutions/src/authentication/providerTokenResponse.ts
/**
 * Copyright(c) Microsoft Corporation.All rights reserved.
 * Licensed under the MIT License.
 */

import { TokenResponse } from 'botframework-schema';
import { OAuthProvider } from './oAuthProvider';

export interface IProviderTokenResponse {
    authenticationProvider: OAuthProvider;
    tokenResponse: TokenResponse;
}

export function isProviderTokenResponse(value?: Object): boolean {
    return value !== undefined && (value as IProviderTokenResponse).authenticationProvider !== undefined;
}
